raft.go

```diff
@@ -1735,6 +1737,7 @@ func (r *raft) handleAppendEntries(m pb.Message) {
 		return
 	}
 	if mlastIndex, ok := r.raftLog.maybeAppend(m.Index, m.LogTerm, m.Commit, m.Entries...); ok {
+		r.raftLog.leaderTerm = m.Term // the log is now consistent with the leader
```

Comment:
@nvanbenschoten This should move to `r.raftLog.maybeAppend`, relatedly to your other comment.

Reply:
Also done in the second commit.